frontend/search/searchController.js

```diff
@@ -28,6 +34,7 @@
         searchCtrl.search = function search() {
             if (searchCtrl.search_keyword) {
                 searchCtrl.makeSearch(searchCtrl.search_keyword, 'institution');
+                searchCtrl.previous_keyword = searchCtrl.search_keyword;
```

Comment:
The controller only calls the backend If the search's string change, otherwise the search is actually a filtering operation into the search result that already exists. So, those two variables controls If there is a change or not in the search's string.